### PR TITLE
feat: 4617 - E2E tests from SVG download2

### PR DESCRIPTION
**Pull request by victorikeokoro6**
## Reason for Change

- #4617
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
- remove
- modify

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

### CODE REVIEW

File: frontend/tests/common/constants.ts [hunk 1]
```diff
@@ -26,3 +26,22 @@ export const BLUEPRINT_SAFE_TYPE_OPTIONS = { delay: 50 };
...
 
 export const LOGIN_STATE_FILENAME = "loginState.json";
+
+export const SHARED_LINK = `${TEST_URL}/gene-expression?compare=disease&datasets=c874f155-9bf9-4928-b821-f52c876b3e48%2Cdb59611b-42de-4035-93aa-1ed39f38b467%2Ceeacb0c1-2217-4cf6-b8ce-1f0fedf1b569%2C881fe679-c6e0-45a3-9427-c4e81be6921f%2Cea786a06-5855-48b7-80d7-0313a21a2044%2C456e8b9b-f872-488b-871d-94534090a865&diseases=MONDO%3A0100096&ethnicities=unknown&sexes=PATO%3A0000383%2CPATO%3A0000384&tissues=blood%2Clung&genes=DPM1%2CTNMD%2CTSPAN6%2CMALAT1&ver=2`;
+
+export const SHARED_LINK_WITHOUT_COMPARE = `${TEST_URL}/gene-expression?datasets=c874f155-9bf9-4928-b821-f52c876b3e48%2Cdb59611b-42de-4035-93aa-1ed39f38b467%2Ceeacb0c1-2217-4cf6-b8ce-1f0fedf1b569%2C881fe679-c6e0-45a3-9427-c4e81be6921f%2Cea786a06-5855-48b7-80d7-0313a21a2044%2C456e8b9b-f872-488b-871d-94534090a865&diseases=MONDO%3A0100096&ethnicities=unknown&sexes=PATO%3A0000383%2CPATO%3A0000384&tissues=blood%2Clung&genes=DPM1%2CTNMD%2CTSPAN6%2CMALAT1&ver=2`;
+
+export const SIMPLE_SHARED_LINK =
+  "https://localhost:3000/gene-expression?tissues=blood%2Clung&genes=DPM1%2CMALAT1%2CTNMD%2CTSPAN6&ver=2";
```
Comment (tihuan) [inline frontend/tests/common/constants.ts:35]: use `TEST_URL`!

File: frontend/tests/features/download/downloadSvg.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      await compareSvg(
+        page,
+        cellSnapshot,
+        geneSnapshot,
+        `${folder}/${tissues[i]}.svg`,
+        folder,
+        tissues[i]
+      );
+      await deleteDownloadedFiles(`./tests/downloads/${folder}`);
+    }
```
Comment (tihuan) [inline frontend/tests/features/download/downloadSvg.test.ts:45]: seems like we can extract common code shared between tests to keep the code DRY?

File: frontend/tests/utils/downloadUtils.ts [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+  //wait for download file
+  const downloadPromise = page.waitForEvent("download");
+
+  //click the download icon
+  await page.getByTestId("download-button").click();
+  const CHECK = "Mui-checked";
+
+  for (const ext of allFileTypes) {
+    const checkboxId = `${ext}-checkbox`;
+    // uncheck if file type is checked but not
```
Comment (tihuan) [inline frontend/tests/utils/downloadUtils.ts:268]: Should we rephrase this comment for clarity?

```suggestion
    // UI has some file types selected by default. This ensures we unselect file types that don't exist in `fileTypes`
```

**Review by tihuan — COMMENTED** [review]
Looks super good! Just some minor comments otherwise LGTM thanks!

File: frontend/tests/common/constants.ts (hunk 1 above)
Comment (victorikeokoro6) [reply frontend/tests/common/constants.ts:35]: Sorry I must have missed this one

**Review by tihuan — APPROVED** [review]
Wow super nice! Thanks so much for refactoring the PR ✨ LGTM!!

**Pull request merged**